### PR TITLE
Added additional parameters to support synced folder and ip-adress

### PR DESCRIPTION
**Pull request by mstellwa**
Added some additional configuration parameters , shared_folder_host and
shared_folder_guest for adding a synced folder and priv_net_ip for
setting the private network ip-adress. Changed the Vagrantfile
accordingly as well.

Also added fix for installing updated vbguest ad-ons.

### CODE REVIEW

**Comment by grtjn** [comment]
Interesting stuff in here, but can you make a few changes?

File: project/Vagrantfile [hunk 1]
```diff
@@ -34,14 +34,24 @@ end
 puts "ml-version=#{@ml_version}"
 puts "nr-hosts=#{@nr_hosts}"
 
+@priv_net_ip = ENV['PRIV_NET_IP'] || @properties['PRIV_NET_IP'] || "192.168.33.10"
```
Comment (grtjn) [inline project/Vagrantfile]: I'd prefer no default ip, and use dhcp as default for private network as well.

File: project/Vagrantfile [hunk 2]
```diff
@@ -53,30 +63,31 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
         #vb.gui = true # for debugging
     end
-    
+
     config.vm.define mastername do |master|
         master.vm.box = "chef/centos-6.5"
         master.vm.provider "virtualbox" do |v|
             v.name = mastername
-            v.memory = 2048
+            v.memory = 6144
```
Comment (grtjn) [inline project/Vagrantfile]: 6gb is too much if you want to create multiple vm's. 2gb per vm is better in that case. Someone already did a pull to make mem and cpu configurable. Time I look into merging that..

File: project/Vagrantfile [hunk 3]
```diff
@@ -53,30 +63,31 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
             v.cpus = 2
         end
         master.vm.hostname = mastername
-        master.vm.network "public_network", bridge: 'en0: Wi-Fi (AirPort)'
+        master.vm.network "private_network", ip: @priv_net_ip
```
Comment (grtjn) [inline project/Vagrantfile]: I'm eager to switch to private network, but with dhcp as default. We should do something like priv_net_ip || dhcp..

File: project/Vagrantfile [hunk 4]
```diff
@@ -34,14 +34,24 @@ end
 puts "ml-version=#{@ml_version}"
 puts "nr-hosts=#{@nr_hosts}"
 
+@priv_net_ip = ENV['PRIV_NET_IP'] || @properties['PRIV_NET_IP'] || "192.168.33.10"
+puts "priv-net-ip=#{@priv_net_ip}"
+
+
+@shared_folder_host = ENV['SHARED_FOLDER_HOST'] || @properties['SHARED_FOLDER_HOST'] || "./"
```
Comment (grtjn) [inline project/Vagrantfile]: ./ = Vagrantfile folder, that is already mounted as /vagrant. I'd suggest no default for this..

File: project/Vagrantfile [hunk 5]
```diff
@@ -34,14 +34,24 @@ end
...
+@shared_folder_guest = ENV['SHARED_FOLDER_GUEST'] || @properties['SHARED_FOLDER_GUEST'] || "/home/vagrant"
```
Comment (grtjn) [inline project/Vagrantfile]: No default for this..

File: project/Vagrantfile [hunk 6]
```diff
@@ -53,30 +63,31 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
+            v.memory = 6144
             v.cpus = 2
         end
         master.vm.hostname = mastername
-        master.vm.network "public_network", bridge: 'en0: Wi-Fi (AirPort)'
+        master.vm.network "private_network", ip: @priv_net_ip
         master.vm.synced_folder "/opt/vagrant", "/opt/vagrant"
         master.vm.synced_folder "/space/software", "/space/software"
-        
+        master.vm.synced_folder @shared_folder_host, @shared_folder_guest, :create => true
```
Comment (grtjn) [inline project/Vagrantfile]: Put a condition around this, if shared_folder then yes, else skip..

File: project/Vagrantfile [hunk 7]
```diff
@@ -91,11 +102,73 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
+
                 data.vm.provision :hostmanager
                 data.vm.provision :shell, :path => "/opt/vagrant/bootstrap-centos-extra.sh", :args => ["-v", "#{@ml_version}", mastername, hostname, File.basename(Dir.getwd)]
             end
         end
     end
-    
+
+end
+module UpdateCentOSKernel
```
Comment (grtjn) [inline project/Vagrantfile]: Interesting stuff, but is this necessary? Maybe there is a newer base vm that we could leverage?

File: project/project.properties [hunk 8]
```diff
@@ -1,2 +1,5 @@
 ml_version=8
-nr_hosts=3
+nr_hosts=1
+priv_net_ip=192.168.33.10
```
Comment (grtjn) [inline project/project.properties]: Put these in as commented entries..

File: project/Vagrantfile (hunk 3 above)
Comment (mstellwa) [inline project/Vagrantfile]: Yes, as long the choice is there.

File: project/project.properties (hunk 8 above)
Comment (mstellwa) [inline project/project.properties]: Sorry, just forgot to reset to the default settings before committing...

File: project/Vagrantfile (hunk 7 above)
Comment (mstellwa) [inline project/Vagrantfile]: Could be. To my understanding this is an issue with CentOS 6.5, could be that another vm or using a newer version of CentOS could solve this.

Comment (grtjn) [inline project/Vagrantfile]: What issue? I have been running CentOS 6.5 vm's for a while, haven't had any trouble..

File: project/project.properties (hunk 8 above)
Comment (grtjn) [inline project/project.properties]: No worries, you can easily append to this PR by committing additional updates to your master branch..

**Comment by mstellwa** [comment]
Added a new commit with changes. Removed the addition to handle kernel update.

**Comment by grtjn** [comment]
Fixed #45 and #24..

**Pull request merged**